SimPEG/Optimization.py

```diff
@@ -990,4 +990,18 @@ def findSearchDirection(self):
                     cgFlag = 1
                 # End CG Iterations
 
+            # Take a gradient step on the active cells if exist
+            if temp != self.xc.size:
```

Comment:
yes, the changes in opt and reg in this pr are behind the sparse reg branch and pr #253. We can do them one after another or combine??